src/sentry/migrations/0203_semver_backfill.py

```diff
@@ -0,0 +1,106 @@
+# Generated by Django 1.11.29 on 2021-05-18 01:44
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+from sentry_relay.exceptions import RelayError
+from sentry_relay.processing import parse_release
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+BATCH_SIZE = 100
+
+
+def convert_build_code_to_build_number(build_code):
+    # Taken from `ReleaseModelManager._convert_build_code_to_build_number`
+    build_number = None
+    if build_code is not None:
+        try:
+            build_code_as_int = int(build_code)
+            if build_code_as_int >= 0 and build_code_as_int.bit_length() <= 63:
+                build_number = build_code_as_int
+        except ValueError:
+            pass
+    return build_number
+
+
+UPDATE_QUERY = """
+    UPDATE sentry_release
+    SET package = data.package,
+    major = data.major,
+    minor = data.minor,
+    patch = data.patch,
+    revision = data.revision,
+    prerelease = data.prerelease,
+    build_code = data.build_code,
+    build_number = data.build_number
+    FROM (VALUES %s) AS data (id, package, major, minor, patch, revision, prerelease, build_code, build_number)
+    WHERE sentry_release.id = data.id"""
+
+
+def backfill_semver(apps, schema_editor):
+    Release = apps.get_model("sentry", "Release")
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        Release.objects.values_list("pk", "version"), result_value_getter=lambda item: item[0]
+    )
+    cursor = connection.cursor()
+    batch = []
+    for pk, version in queryset:
+        try:
+            version_info = parse_release(version)
+        except RelayError:
+            continue
+
+        version_parsed = version_info.get("version_parsed")
+        if version_parsed is None:
+            continue
+
+        build_code = version_parsed.get("build_code")
+        build_number = convert_build_code_to_build_number(build_code)
+        batch.append(
+            (
+                pk,
+                version_info["package"],
+                version_parsed["major"],
+                version_parsed["minor"],
+                version_parsed["patch"],
+                version_parsed["revision"],
+                version_parsed["pre"] or "",
+                build_code,
+                build_number,
+            )
+        )
+        if len(batch) >= BATCH_SIZE:
+            execute_values(cursor, UPDATE_QUERY, batch, page_size=BATCH_SIZE)
+            batch = []
+
+    if batch:
+        execute_values(cursor, UPDATE_QUERY, batch, page_size=BATCH_SIZE)
```

Comment:
Could this last batch be oddly sized? Should `page_size` be `len(batch)`?

Reply:
It doesn't matter too much, this is essentially a ceiling on how many things are executed.